core/src/databases.rs

```diff
@@ -18,22 +18,47 @@ impl MongoDatabases {
     // All columns except the TABLE_CAT column contain NULLs.
     // The query timeout comes from the statement attribute SQL_ATTR_QUERY_TIMEOUT. If there is a
     // timeout, the query must finish before the timeout or an error is returned.
-    pub fn list_all_catalogs(_client: &MongoConnection, _query_timeout: Option<i32>) -> Self {
-        unimplemented!()
+    pub fn list_all_catalogs(
+        mongo_connection: &MongoConnection,
+        _query_timeout: Option<i32>,
+    ) -> Self {
+        let database_names: Vec<String> = mongo_connection
+            .client
+            .list_database_names(None, None)
+            .unwrap();
+        MongoDatabases {
+            database_names,
+            current_db_index: 0,
+        }
     }
 }
 
 impl MongoStatement for MongoDatabases {
     // Increment current_db_index.
     // Return true if current_db_index index is < for databases_names.length.
     fn next(&mut self) -> Result<bool> {
-        unimplemented!()
+        let current = self.current_db_index;
+        self.current_db_index += 1;
+        Ok(current < self.database_names.len())
     }
 
     // Get the BSON value for the value at the given colIndex on the current row.
-    fn get_value(&self, _col_index: u16) -> Result<Option<&Bson>> {
+    fn get_value(&self, col_index: u16) -> Result<Option<Bson>> {
         // The mapping for col_index <-> Value will be hard-coded and handled in this function
         // 1-> databases_names[current_row_index]
-        unimplemented!()
+        match col_index {
+            1 => Ok(Some(Bson::String(
+                self.database_names
+                    .get(self.current_db_index)
```

Comment:
The flow is list_all_catalogs, (next, getValue*)*.
You'll need to use `self.current_db_index - 1`
Otherwise, it skips the first value and will fail after the last successful call to next.